src/components/UploadCSVView/CSVViewTable/CSVViewTable.jsx

```diff
@@ -41,93 +149,147 @@ const CSVViewTable = ({ rows }) => {
+        const allCoordinates = await Promise.all(
+          formDatas.map(async formData => getLatLong(formData.zipCode, formData.country)),
+        );
+
+        let hasError = false;
+        formDatas.forEach((formData, index) => {
+          if (allCoordinates[index].length === 0) {
+            // if lat/long is not found for this zipcode
+            console.log('cannot find latitude for formData: ', formData);
+            hasError = true;
+            setIsLoading(false);
+            editRow(e, formData, index, false);
+          } else {
+            // otherwise, set lat/long for this zipcode
+            const [lat, long] = allCoordinates[index];
+            formDatas[index].latitude = lat;
+            formDatas[index].longitude = long;
+          }
+        });
+
+        // set lat/long for each formData
+        // formDatas.forEach((formData, index) => {
+        //   const [lat, long] = allCoordinates[index];
+        //   formDatas[index].latitude = lat;
+        //   formDatas[index].longitude = long;
+        // });
+
+        // if none of the rows have any errors
+        if (!hasError) {
+          await FYABackend.post('/anchorBox/boxes', formDatas);
+          setIsLoading(false);
+          navigate('/admin');
+        }
+      } catch (err) {
+        console.log(err);
+      }
     }
   };
 
-  // check if any rows have an error when page first loads
-  useEffect(async () => {
-    setFormData(rows);
-    const errors = await checkErrors(rows);
-    addErrors(errors);
-  }, [rows]);
-
   return (
     <form onSubmit={addToMap} className={styles['csv-table-form']}>
+      <Stack direction="row" justify="right" marginTop="-25px" marginBottom="25px">
+        <select
+          value={pageSize}
+          onChange={e => {
+            setPageSize(Number(e.target.value));
+          }}
+          className={styles['show-pages-select']}
+        >
+          {[10, 20, 30, 40, 50].map(pageSizeVal => (
+            <option key={pageSizeVal} value={pageSizeVal}>
+              Show {pageSizeVal}
+            </option>
+          ))}
+        </select>
+        <Button isLoading={isLoading} type="submit" color="white" bg="#345E80">
+          Add to Map
+        </Button>
+      </Stack>
       <div className={`${styles['csv-table-container']} ${styles['scrollable-div']}`}>
         <Table className={styles['csv-table']}>
           <Thead>
             <Tr>
               <Th>Date</Th>
               <Th>Box Number</Th>
               <Th>Zip Code</Th>
+              <Th>Country</Th>
               <Th>Launched Organically</Th>
               <Th>&nbsp;</Th>
             </Tr>
           </Thead>
           <Tbody>
-            {formDatas.map(data => {
+            {page.map((rowData, index) => {
+              prepareRow(rowData);
               return (
-                <Fragment key={data.id}>
-                  {editId === data.id ? (
+                <Fragment key={rowData.original.id}>
```

Comment:
Not sure if this is the right area to look in but when I upload a CSV file with no errors and then click on "Edit/View File", I'm not able to edit any of the rows. When I click on the pencil icon, it changes to an editable row, waits 1 second, and then reverts back to uneditable row. Could y'all look into why this is happening?

Reply:
For more context - I uploaded a CSV file with no errors except the lat/long couldn't be found

Reply:
Seems like this actually happens any time the row has no errors and the user clicks the pencil icon. For this case, the user should still be able to edit the row even if there are no errors.